src/coreclr/nativeaot/Runtime/threadstore.inl

```diff
@@ -1,7 +1,12 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
-EXTERN_C DECLSPEC_THREAD ThreadBuffer tls_CurrentThread;
+#ifdef _MSC_VER
+// a workaround to prevent tls_CurrentThread from becoming dynamically checked/initialized.
```

Comment:
Can we keep on `extern "C"` on this workaround to avoid introducing name mangling into the asm code?

Reply:
I just ported the workaround form a similar change in CoreClr. I am not sure I completely understand why it is needed. 
If removing extern "C" is not a part of the fix, I guess, why not. I will try.

Reply:
The fix work with extern "C", so I will add it and revert the mangling in asm